### PR TITLE
Set enclave load policy from env var

### PR DESCRIPTION
**Pull request by asvrada**
Also Fix intel_qve_thread_test not returning error if threads error out

Reference:
* https://github.com/openenclave/openenclave/compare/master...asvrada:openenclave:qvl_load_policy
* https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/6882afad8644c27db162b40994402c8ad2a7fb32/QuoteGeneration/pce_wrapper/inc/sgx_pce.h#L80

### CODE REVIEW

**Comment by asvrada** [comment]
bors try

File: host/sgx/sgxquote.c [hunk 1]
```diff
@@ -170,6 +176,50 @@ typedef struct _supp_ver_t
...
 
+// Starting from
+// https://github.com/intel/SGXDataCenterAttestationPrimitives/releases/tag/DCAP_1.18
+// The API (sgx_qv_set_enclave_load_policy) is enhanced to support changing QvL
+// multithreading behavior At runtime, OE read env var
+// "OE_INTEL_QVL_LOAD_POLICY=?" to set sgx_ql_request_policy_t policy Accepted
+// values: SGX_QL_PERSISTENT, SGX_QL_EPHEMERAL,
+// SGX_QL_EPHEMERAL_QVE_MULTI_THREAD, SGX_QL_PERSISTENT_QVE_MULTI_THREAD,
+// SGX_QL_DEFAULT
+static sgx_ql_request_policy_t _env2load_policy(void)
```
Comment (mingweishih) [inline host/sgx/sgxquote.c:187]: Can we have better name for this function (e.g., `_get_qvl_load_policy`)

File: 3rdparty/sgxsdk/include/sgx_dcap_quoteverify.h [hunk 2]
```diff
@@ -49,18 +49,23 @@ extern "C" {
  * When the Quoting Verification Library is linked to a process, it needs to know the proper enclave loading policy.
  * The library may be linked with a long lived process, such as a service, where it can load the enclaves and leave
  * them loaded (persistent). This better ensures that the enclaves will be available upon quote requests and not subject
- * to EPC limitations if loaded on demand. However, if the Quoting library is linked with an application process, there
- * may be many applications with the Quoting library and a better utilization of EPC is to load and unloaded the quoting
- * enclaves on demand (ephemeral).  The library will be shipped with a default policy of loading enclaves and leaving
- * them loaded until the library is unloaded (PERSISTENT). If the policy is set to EPHEMERAL, then the QE and PCE will
- * be loaded and unloaded on-demand.  If either enclave is already loaded when the policy is change to EPHEMERAL, the
- * enclaves will be unloaded before returning.
+ * to EPC limitations if loaded on demand. However, if the QVL is linked with an application process, there may be many
```
Comment (radhikaj) [inline 3rdparty/sgxsdk/include/sgx_dcap_quoteverify.h:52]: can you add to changelog?

File: host/sgx/sgxquote.c [hunk 3]
```diff
@@ -553,6 +626,10 @@ static void _load_sgx_dcap_qvl_impl(void)
 
     if (_qvl_module)
     {
+        OE_CHECK(_lookup_function(
+            _qvl_module,
+            "sgx_qv_set_enclave_load_policy",
+            (void**)&_sgx_qv_set_enclave_load_policy));
```
Comment (radhikaj) [inline host/sgx/sgxquote.c:632]: why does this have an _sgx_ when the next looked up funciton pointer just has "sgx_"

**Review by radhikaj — COMMENTED** [review]
Looks good overall @asvrada. Just looking for some naming changes

File: host/sgx/sgxquote.c (hunk 3 above)
Comment (asvrada) [reply host/sgx/sgxquote.c:632]: To add support for both SGX (here) and TDX(next lookup function)

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Review by mingweishih — APPROVED** [review]

**Review by radhikaj — APPROVED** [review]

**Pull request merged**